spec/lib/wolf_waagen_api/import_spec.rb

```diff
@@ -0,0 +1,250 @@
+      }
+    end
+
+    context 'given sensor for other reports' do
+      before { other_type_sensor }
+
+      it { is_expected.not_to include(other_report_sensor) }
+    end
+
+    context 'given no matching sensor' do
+      before { other_sensor_type }
+
+      it { is_expected.to eq([]) }
+    end
+
+    context 'given no sensor types match' do
+      it { is_expected.to eq([]) }
+    end
+  end
+
+  describe '#api_sensors_by_series_id' do
+    subject do
+      specified_type = WolfWaagenApi::Import::SENSOR_TYPES.first
+      import = described_class.new(report: report)
+      import.api_sensors_by_series_id(series_id: specified_type[:series_id])
+    end
+
+    context 'if no sensor type matches' do
+      before do
+        other_type_sensor
+      end
+
+      it { is_expected.to eq([]) }
+    end
+
+    context 'if no sensor matches' do
+      before do
+        wolf_sensor_type
+      end
+
+      it { is_expected.to eq([]) }
+    end
+  end
+
+  describe '#start_date' do
+    subject { described_class.new(report: report).start_date }
+
+    context 'given no sensor readings for one of two sensors' do
+      before do
+        wolf_sensor
+        wolf_sensor2
+        wolf_sensor_reading
+      end
+
+      it 'returns the report start date' do
+        is_expected.to eq(DateTime.new(2019, 1, 1))
+      end
+    end
+
+    context 'given sensor readings for all sensors' do
+      before do
+        wolf_sensor
+        wolf_sensor2
+        wolf_sensor_reading
+        wolf_sensor_reading2
+      end
+
+      it 'returns the oldest latest sensor reading date' do
+        is_expected.to eq(DateTime.new(2019, 1, 2))
+      end
+    end
+  end
+
+  describe '#save_series_points' do
+    let(:request) do
+      WolfWaagenApi::Request.new(
+        hive_id: 'HIVE_ID',
+        start_date: DateTime.new(2019, 1, 1, 0, 0),
+        end_date: DateTime.new(2019, 1, 2, 0, 0)
+      )
+    end
+
+    let(:result) do
+      WolfWaagenApi::Result.new(
+        request: request,
+        data: {
+          pointStart: 1546300800000, # 2019-01-01 00:00:00
+          pointInterval: 60 * 60 * 1000,
+          series: [{
+            id: WolfWaagenApi::Import::SENSOR_TYPES.first[:series_id],
+            values: [10, 20, 30]
+          }]
+        }
+      )
+    end
+
+    let(:import) do
+      described_class.new(report: report)
+    end
+
+    context 'given a Wolf Waagen sensor' do
+      before { wolf_sensor }
+
+      it 'creates sensor readings for that sensor' do
+        import.save_series_points(series: result.series.first)
+        saved_values = wolf_sensor.sensor_readings.pluck(:calibrated_value)
+
+        expect(saved_values).to eq([10, 20, 30])
+      end
+
+      it 'does not create duplicate sensor readings' do
+        import.save_series_points(series: result.series.first)
+        import.save_series_points(series: result.series.first)
+        saved_values = wolf_sensor.sensor_readings.pluck(:calibrated_value)
```

Comment:
Make `saved_values`your subject
```ruby
subject {  wolf_sensor.sensor_readings.pluck(:calibrated_value) }
```
put the setup into a `before` respectively, e.g:
```
before do
        import.save_series_points(series: result.series.first)
        import.save_series_points(series: result.series.first)
end
```
and have a nice one-liner for the test cases:
```
it { is_expected.to eq([10, 20, 30]) }
```